tests/js/async/await-keyword.js

```diff
@@ -0,0 +1,5 @@
+async function foo() {
+  function bar(x = await 2) {}
+}
+
+async (x = await 2) => {};
```

Comment:
Can you also test, Note that `flow` parse this differently.

**Prettier pr-10345**
[Playground link](https://deploy-preview-10345--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBDAzgTymABAMwFccYBLaAiCACgEpdgAdKXA4sMigI1QCdqAHrgC8uVAHdUpGLmoAmWvWABfZqqjMM2PIJFjJ02QvrCAfA2UBuEABoQEAA6co6ZKD68I4gAp8ErlFQAG0lMVzsuXlQwAGs4GABlB2jSKABzZBheQjg7OABbLjgAE2KSgBlUdMJUNLgAMQhefNQYMnTkNEIYCFsQAAsYfKCAdX7pOHRksDgE-2lSADdpTE6wdHCQVPQ4XhhvKLSW5HxgnbsAK3QBACEo2PiE1Hy4ctS4E7PckCuBBNS0kE4ABFQgQeCfILnEDJXg7XidHhFIJ9By8VIwEakYowfrIAAcAAY7GiIDsRlEHJ00ZNdosPnYAI5g+AHRwBNDoAC0UDgJRKfV4cGZpCFB1qxyQpyh3x2+VImWysoBQNB4I+Uq+dhgqC4WJxeKQcm1UVIQQBAGEIPlJSBJgBWPqEHYAFV1AWl0MWOQAklAyrAEmB0U4AIL+hIwTBAyE7ZTKIA)
<!-- prettier-ignore -->
```sh
--parser babel
```

**Input:**
<!-- prettier-ignore -->
```jsx
async function foo() {
  function bar(x = await (2)) {}
}

async (x = await (2)) => {};
```

**Output:**
<!-- prettier-ignore -->
```jsx
async function foo() {
  function bar(x = await(2)) {}
}

async (x = await 2) => {};

```